README.md

```diff
@@ -109,6 +109,43 @@ server := Server{
 log.Fatal(http.ListenAndServe(":8080", server))
 ```
 
+### Filtering
+In the `ListRequestParams` a parsed `Filter` can be found. This is just the filter query from the incoming request.
+This structure provides a two methods:
+
+- `Reduce`: removes all resources from a []map that do not pass the filter.
+- `IsValid`: checks whether the given map is passed the filter.
```

Comment:
Is there a specific reason for designing the API like this? Would it also be possible to just filter the resources before they reach the handlers, and to not provide a parsed filter at all?

Reply:
I do not want to force people to use this implementation.
I think it is much more interesting to write a transpiler ([example](github.com/articulate/scim-filter-transpiler)).

Reply:
Filtering beforehand can also create conflicts w/ paging.

Reply:
I see, but IMO we should try to keep the public API as small and simple as possible. If we don't want to implement actual filtering ourselves (which is indeed a good idea), we should just expose an AST of the parsed filter without the additional helper methods.

Reply:
I will keep it internal for now. If requested we can think about a better public API.